tests/compiler/expressions.t/apply.ml

```diff
@@ -32,3 +32,11 @@ let lambda () =
   let _ = f () in
   let _ = f' 1 in
   f'' 1 2
+
+let abs_diff_times x y z = (Erlang.abs (x - y)) * z
+
+let partial_application () =
+  let f = abs_diff_times 3 in
+  f (-1) 2
+
+let another_partial _ = Io.format "Hello"
```

Comment:
What happens if you do

```ocaml
let top_level_partial () = Io.format "hello"
```

or

```ocaml
let top_level_partial _ = Io.format "hello"
```

Reply:
2nd one "cannot be generalized"
1st one has the same result as with `_`, so
```
  -spec top_level_partial(ok, list(_)) -> ok.
```
instead of
```
  -spec top_level_partial() -> fun(list(_) -> ok).
```

Reply:
(I just noticed my failing test is wrong, it should say what I typed above instead of just `-spec name(_) -> ok.`